tests/ui/coherence/warn-when-cycle-is-error-in-coherence.stderr

```diff
@@ -0,0 +1,22 @@
+error: impls that are not considered to overlap may be considered to overlap in the future
+  --> $DIR/warn-when-cycle-is-error-in-coherence.rs:13:1
+   |
+LL | #[derive(PartialEq, Default)]
+   |          --------- the second impl is here
+...
+LL | impl<T, Q> PartialEq<Q> for Interval<T>
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the first impl is here
+...
+LL |     Q: ?Sized + PartialOrd,
+   |                 ---------- this where-clause causes a cycle, but it may be treated as possibly holding in the future, causing the impls to overlap
```

Comment:
It would be nice if we mention the predicates after resolving the vars.
```suggestion
error: conflicting implementations for `Interval<_>: PartialEq<Interval<_>>`
  --> $DIR/warn-when-cycle-is-error-in-coherence.rs:13:1
   |
LL | #[derive(PartialEq, Default)]
   |          --------- the second impl is here
...
LL | impl<T, Q> PartialEq<Q> for Interval<T>
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the first impl is here
...
LL |     Q: ?Sized + PartialOrd,
   |                 ---------- `Interval<_>: PartialOrd<Interval<_>>` will be considered to hold in future releases, causing the impls to overlap
```